### PR TITLE
[MIG][12.0] agreement_project migration

### PR DESCRIPTION
**Pull request by ygol**
Part of #189

### CODE REVIEW

**Review by max3903 — APPROVED** [review]
LGTM

File: agreement_project/README.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    :alt: smangukiya
+.. |maintainer-max3903| image:: https://github.com/max3903.png?size=40px
+    :target: https://github.com/max3903
+    :alt: max3903
+
+Current `maintainers <https://odoo-community.org/page/maintainer-role>`__:
+
+|maintainer-smangukiya| |maintainer-max3903| 
+
+This module is part of the `OCA/contract <https://github.com/OCA/contract/tree/11.0/agreement_project>`_ project on GitHub.
```
Comment (murtuzasaleh) [inline agreement_project/README.rst]: 12.0

File: agreement_project/README.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fcontract-lightgray.png?logo=github
+    :target: https://github.com/OCA/contract/tree/11.0/agreement_project
+    :alt: OCA/contract
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-agreement_project
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/110/11.0
```
Comment (murtuzasaleh) [inline agreement_project/README.rst]: 12.0

File: agreement_project/README.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fcontract-lightgray.png?logo=github
+    :target: https://github.com/OCA/contract/tree/11.0/agreement_project
```
Comment (murtuzasaleh) [inline agreement_project/README.rst]: 12.0

File: agreement_project/README.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    :alt: OCA/contract
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-agreement_project
```
Comment (murtuzasaleh) [inline agreement_project/README.rst]: 12

Comment (ygol) [reply agreement_project/README.rst]: @max3903 can you confirm that the README.rst is automatically generated/updated by the maintainer-tools?
@murtuzasaleh If the above is correct the README.rst file will be magically updated

Comment (nikul-serpentcs) [reply agreement_project/README.rst]: Yes @ygol , But implicit better than explicit, So here we can done version number also.

**Review by murtuzasaleh — APPROVED** [review]

File: agreement_project/models/agreement.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+class Agreement(models.Model):
+    _inherit = "agreement"
+
+    task_count = fields.Integer('# Tasks',
+                                compute='_compute_task_count')
+
+    @api.multi
+    def _compute_task_count(self):
+        data = self.env['project.task'].read_group(
```
Comment (nikul-serpentcs) [inline agreement_project/models/agreement.py]: @ygol  I think here use `search_count` method, then Direct Get count number

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improved Code

**Comment by nikul-serpentcs** [comment]
@ygol Squash Commit & rebase

**Comment by ygol** [comment]
> 
> 
> @ygol Squash Commit & rebase

explain

**Comment by nikul-serpentcs** [comment]
> > @ygol Squash Commit & rebase
> 
> explain

@ygol  Squash last 2 commits and rebase your branch

**Review by osi-scampbell — APPROVED** [review]

File: agreement_project/models/agreement.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    _inherit = "agreement"
+
+    task_count = fields.Integer('# Tasks',
+                                compute='_compute_task_count')
+
+    @api.multi
+    def _compute_task_count(self):
+        for ag in self:
+            count = self.env['project.task'].search_count(
+                [('agreement_id', 'in', self.ids)])
```
Comment (nikul-serpentcs) [inline agreement_project/models/agreement.py:17]: ` [('agreement_id', 'in', ag.ids)]` or `[('agreement_id', '=', ag.id)]`

@ygol  @max3903 `self.ids` is correct it's call one time in form view, but if we used `for loop` then use loop variable.

Comment (max3903) [reply agreement_project/models/agreement.py:17]: @nikul-serpentcs Can you make a pull request please?

**Pull request merged**